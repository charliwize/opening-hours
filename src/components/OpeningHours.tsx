import styled from 'styled-components';
import OpeningHour from './OpeningHour';

interface Props {
    scheduleInfo: { day: string; schedule: string[] }[];
}

const Times = styled.div`
    padding: 1.5rem;
    margin-top: 2.5rem;
    box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
    border-radius: 10px;

    .spinner-border {
        display: flex;
        margin: 1.5rem auto;
    }
`;

const Header = styled.div`
    border-bottom: 1.5px solid #202125;
    padding-bottom: 1rem;
    display: flex;
    h2 {
        margin: 0;
        font-weight: bold;
    }
    img {
        width: 18px;
        margin-right: 0.5rem;
    }
`;

const EmptyState = styled.div`
    margin-top: 1.5rem;
`;

const WeekSchedule = ({ scheduleInfo }: Props): React.ReactElement => {
    if (!Boolean(scheduleInfo.length)) {
        return <EmptyState>No available opening hours</EmptyState>;
    }

    return (
        <>
            {scheduleInfo.map((item, index) => {
                return <OpeningHour item={item} key={index} />;
            })}
        </>
    );
};

const OpeningHours = ({ scheduleInfo }: Props) => {
    return (
        <Times>
            <Header>
                <img src="/assets/clock-svgrepo-com.svg" alt="clock" />
                <h2>Opening Hours</h2>
            </Header>
            <WeekSchedule scheduleInfo={scheduleInfo} />
        </Times>
    );
};

export default OpeningHours;
