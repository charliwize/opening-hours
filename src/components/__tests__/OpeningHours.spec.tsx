import { render, screen } from '@testing-library/react';
import React from 'react';
import OpeningHours from '../OpeningHours';

const defaultProps = {
    items: [
        {
            day: 'Saturday',
            schedule: ['9 AM - 11 AM', '4 PM - 11 PM'],
        },
        {
            day: 'saturday',
            schedule: ['9 AM - 11 AM', '4 PM - 11 PM'],
        },
    ],
};

describe('OpeningHours', () => {
    test('it renders', () => {
        const { container } = render(<OpeningHours scheduleInfo={defaultProps.items} />);
        expect(screen.getByText('Opening Hours')).toBeTruthy();
        expect(container).toMatchSnapshot();
    });

    test('it should render no opening hours', () => {
        render(<OpeningHours scheduleInfo={[]} />);
        expect(screen.getByText('No available opening hours')).toBeTruthy();
    });

    test('it should render opening hours', () => {
        render(<OpeningHours scheduleInfo={defaultProps.items} />);
        expect(screen.getAllByTestId('opening-hour')).toHaveLength(2);
    });
});
