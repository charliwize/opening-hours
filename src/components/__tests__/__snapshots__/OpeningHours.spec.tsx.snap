// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningHours it renders 1`] = `
<div>
  <div
    class="sc-hKwDye fQBrgi"
  >
    <div
      class="sc-eCImPb cvQtDW"
    >
      <img
        alt="clock"
        src="/assets/clock-svgrepo-com.svg"
      />
      <h2>
        Opening Hours
      </h2>
    </div>
    <div
      class="sc-bdvvtL iILrCD"
      data-testid="opening-hour"
    >
      <div>
        <strong
          class="day-name"
        >
          Saturday
        </strong>
      </div>
      <div
        class="sc-dkPtRN hcoMhN"
      >
        <div
          class=""
        >
          9 AM - 11 AM
        </div>
        <div
          class=""
        >
          4 PM - 11 PM
        </div>
      </div>
    </div>
    <div
      class="sc-bdvvtL iILrCD"
      data-testid="opening-hour"
    >
      <div>
        <strong
          class="day-name"
        >
          saturday
        </strong>
      </div>
      <div
        class="sc-dkPtRN hcoMhN"
      >
        <div
          class=""
        >
          9 AM - 11 AM
        </div>
        <div
          class=""
        >
          4 PM - 11 PM
        </div>
      </div>
    </div>
  </div>
</div>
`;
